images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -250,12 +250,25 @@ $(generate_script_config "${ipaddr}" "${port}")
 
   local counter=1
   local previous="none"
+  local i=0
+  local vip_counter=0
+  local total_vips=( $vips )
+  local vips_per_group=1
+  local vips_mod=0
+
+  if [[ $vip_groups -ne 0 ]]; then
+    vips_per_group=$((${#total_vips[@]} / vip_groups))
+    vips_mod=$((${#total_vips[@]} % vip_groups))
+  fi
 
-  for vip in ${vips}; do
+  while [[ "${total_vips[$vip_counter]}" != "" ]]; do
```

Comment:
offset, priority, instancetype variables need to be generated only once for  each call to generate_vrrpd_instance_config() (i.e. once for each group). 
It will be cleaner and easier if you can abstract vips per group generation in a separate method.
```
function generate_vip_groups() {
...
// return nested associate array 
// eg: [ ["1.2.3.4", "1.2.3.5"], ["1.2.3.6", "1.2.3.7"], ["1.2.3.8"]] 
}

// Instead of  'for vip in ${vips}; do', now it will be
for vip_group in  generate_vip_groups; do
...
```

```

Reply:
bash does not support this type of array, i'll leave this part as it is

Reply:
I think we can workaround with out this array support. 
We could use https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion and this will simplify the code as well.

```
#!/bin/bash

function test_sample() {
    local total_vips=(1 2 3 4 5 6 7 8)
    local vip_counter=0
    local vips_per_group=3

    while [[ "${vip_counter}" -le "${#total_vips[@]}" ]]; do

        vip_group=("${total_vips[@]:vip_counter:vips_per_group}")
        vip_counter=$((vip_counter + vips_per_group))

        echo "VIP group: ${vip_group[*]}"

        #
        # Process each vip group
        #
    done
}

test_sample
```